poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -283,26 +281,24 @@ public void removeSensorMetaData(FacilitySoftwareConfig powerIQ) {
       List<Asset> needToUpdate = new ArrayList<Asset>();
       for(Asset server : servers) {
          boolean changed = false;
-         Map<ServerSensorType,String> sensorsformular = server.getSensorsformulars();
-         if(sensorsformular.isEmpty()) {
+         Map<String, Map<String, Map<String, String>>> formulars = server.getMetricsformulars();
+         if(formulars == null || formulars.isEmpty()) {
             continue;
          }
-         String humidity = sensorsformular.get(ServerSensorType.HUMIDITY);
-         if(humidity != null && humidity.indexOf(sensorId) >= 0) {
-            sensorsformular.remove(ServerSensorType.HUMIDITY);
-            changed = true;
-         }
-         String frontPanelTemp = sensorsformular.get(ServerSensorType.FRONTPANELTEMP);
-         if(frontPanelTemp != null && frontPanelTemp.indexOf(sensorId) >= 0) {
-            sensorsformular.remove(ServerSensorType.FRONTPANELTEMP);
-            changed = true;
-         }
-         String backPanelTemp = sensorsformular.get(ServerSensorType.BACKPANELTEMP);
-         if(backPanelTemp != null && backPanelTemp.indexOf(sensorId) >= 0) {
-            sensorsformular.remove(ServerSensorType.BACKPANELTEMP);
-            changed = true;
+         Map<String, Map<String, String>> sensorFormulars = formulars.get(FlowgateConstant.SENSOR);
+         for(Map.Entry<String, Map<String, String>> sensorFormularMap : sensorFormulars.entrySet()) {
+            Map<String, String> sensorLocationAndIdMap = sensorFormularMap.getValue();
+            Iterator<Map.Entry<String, String>> ite = sensorLocationAndIdMap.entrySet().iterator();
+            while(ite.hasNext()) {
+               Map.Entry<String, String> map = ite.next();
+               String sensorIdFromServer = map.getValue();
+               if(sensorIdFromServer.equals(sensorId)) {
```

Comment:
this not right, you should only remove the assetid instead of the remove the formula. unless it is the last asset id in the formula.